manifests/satosa.pp

```diff
@@ -1,79 +1,121 @@
+        cert_file => "/etc/satosa/${id}.crt"
       }
-   }
-   ufw::allow { "satosa-allow-https":
-      ip   => 'any',
+    }
+  }
+  file {'/etc/satosa/proxy_conf.yaml':
+    content => inline_template("<%= @merged_conf.to_yaml %>\n"),
+    notify  => Service['sunet-satosa'],
+  }
+  $plugins = hiera('satosa_config')
+  sort(keys($plugins)).each |$n| {
+    $conf = hiera($n)
+    $fn = $plugins[$n]
+    file { $fn:
+      content => inline_template("<%= @conf.to_yaml %>\n"),
+      notify  => Service['sunet-satosa'],
+    }
+  }
+
+  if $::facts['sunet_nftables_enabled'] == 'yes' {
+    sunet::nftables::docker_expose { 'allow_https' :
+      iif           => $interface,
+      allow_clients => 'any',
+      port          => 443,
+    }
+  } else {
+    sunet::misc::ufw_allow { 'allow-https':
+      from => 'any',
       port => '443'
-   }
-   $dehydrated_status = $dehydrated_name ? {
-      undef   => 'absent',
-      default => 'present'
-   }
-   sunet::docker_run {'alwayshttps':
-      image    => 'docker.sunet.se/always-https',
-      ports    => ['80:80'],
-      env      => ['ACME_URL=http://acme-c.sunet.se'],
-      ensure   => $dehydrated_status
-   }
-   ufw::allow { "satosa-allow-http":
-      ip     => 'any',
-      port   => '80',
-      ensure => $dehydrated_status
-   }
-   if ($dehydrated_name) {
-      file { '/etc/satosa/https.key': ensure => link, target => "/etc/dehydrated/certs/$dehydrated_name.key" }
-      file { '/etc/satosa/https.crt': ensure => link, target => "/etc/dehydrated/certs/${dehydrated_name}/fullchain.pem" }
-   } else {
-      sunet::snippets::keygen {"satosa_https":
-         key_file  => "/etc/satosa/https.key",
-         cert_file => "/etc/satosa/https.crt"
+    }
+  }
+  $dehydrated_status = $dehydrated_name ? {
+    undef   => 'absent',
+    default => 'present'
+  }
+
+  if ($dehydrated_name) {
+    class { 'sunet::dehydrated::client':
+      domain    => $dehydrated_name,
+      ssl_links => true,
+    }
+
+    if $::facts['sunet_nftables_enabled'] == 'yes' {
+      sunet::nftables::docker_expose { 'allow_http' :
+        iif           => $interface,
+        allow_clients => 'any',
+        port          => 80,
+      }
+    } else {
+      sunet::misc::ufw_allow { 'allow-http':
+        from => 'any',
+        port => '80'
       }
-   }
+    }
+    file { '/etc/satosa/https.key': ensure => link, target => "/etc/dehydrated/certs/${dehydrated_name}.key" }
+    file { '/etc/satosa/https.crt': ensure => link, target => "/etc/dehydrated/certs/${dehydrated_name}/fullchain.pem" }
+  } else {
+    sunet::snippets::keygen {'satosa_https':
+      key_file  => '/etc/satosa/https.key',
+      cert_file => '/etc/satosa/https.crt'
+    }
+  }
+
+  if ($enable_oidc) {
+    $client_id     = lookup('client_id')
+    $client_secret = lookup('client_secret')
+    file { '/etc/satosa/cdb.json':
+      ensure  => file,
+      content => template('sunet/satosa/cdb.json.erb')
```

Comment:
Should we make it possible to add your own cdb.json here? Possibly by adding a "manage_cdb" parameter to the class?

Reply:
I will try to completely rewrite this segment with you comment in mind. Stay tuned.

Reply:
A few commits later and the json configurations are now handled in the same way as all other yaml configuration in this class. Think it got a lot better.

Please have another look.